tiledb/ml/data_apis/tensorflow_sparse.py

```diff
@@ -164,15 +205,16 @@ def _generator_sparse_dense(
                 dim_name = x.schema.domain.dim(i).name
                 x_coords.append(np.array(x_batch[dim_name]))
 
-            x_data = np.array(values_x).flatten()
+            # Normalise indices for torch.sparse.Tensor We want the coords indices in every iteration
+            # to be in the range of [0, self.batch_size] so the torch.sparse.Tensors can be created batch-wise.
+            # If we do not normalise the sparse tensor is being created but with a dimension [0, max(coord_index)],
+            # which is overkill
+            x_coords[0] -= x_coords[0].min()
 
-            # Detects empty lines due to sparsity inside the batch
-            x_sparse_real_batch_size = x_coords[0].max(axis=0) - x_coords[0].min(axis=0)
-            if values_y.shape[0] - x_sparse_real_batch_size != 1:
-                raise ValueError(
-                    "X and Y should have the same number of rows, i.e., the 1st dimension "
-                    "of TileDB arrays X, Y should be of equal domain extent inside the batch"
-                )
+            # for the check slice the row dimension of y dense array
+            cls._check_row_dims(x_coords[0], values_y)
```

Comment:
Does this work? `values_y` is a list, the check function expects a ndarray. 

Regardless of this, can `values_x`/`values_y` become ndarrays instead of lists to save an extra allocation and conversion to `x_data/`y_data`?

Reply:
I have made the `x_batch.items()` a list in order to abstract from the need of giving hard-coded keys for the ordered dict in order to take the values. It is still with indices that correspond to the values field but I could not think of sth more general than that. However getting these values are returned as nd.array so `values_y` are `nd.array`.

Reply:
```
values_y = list(y_batch.items())[0][1]
...
# values_y is not reassigned 
....
cls.__check_row_dims(x_coords[0], values_y, sparse=False)
```
How can `values_y` be an array?